proxy/src/coco/peer.rs

```diff
@@ -14,10 +14,10 @@ pub use librad::net::peer::{PeerApi, PeerConfig};
 use librad::paths;
 use librad::peer::PeerId;
 use librad::uri::RadUrn;
-use radicle_surf::vcs::git::{self, git2};
+use radicle_surf::vcs::git::{self, git2, Stats};
 
 use crate::error;
-use crate::project::Project;
+use crate::project::{DiscoveryItem, Metadata, Project};
```

Comment:
Continue to importing this is violating our domain boundaries. This module should not be aware of the internals of the project/discovery domain. I'd like to see this addressed for the reason that if we don't do it now it will start to creep and we further blurr these boundaries. I understand that there is a clear formaulation of how these boundaries look like and will work to provide appropriate documentation for it.

Reply:
For interacting with the `peer_api`, would you suggest something like
```
    pub fn get_entity(&self, urn: &RadUrn) -> Result<entity::Entity, error::Error> {
        let api = self.peer_api.lock().expect("unable to accquire lock");
        let storage = api.storage().reopen()?;

        Ok(storage.metadata(urn)?)
    }
```

instead of the existing `get_project` and `get_user` functions? Or would it make more sense to expose the `peer_api` & `storage` to `project`/`user`/other proxy modules?